### DISCUSSION TITLE
No NDR mail is sent to the return path if no servers are available. HARD BOUNCE

### DISCUSSION THREAD

**Discussion by odoo-mvds** [postalserver/postal#3100] [Help with using Postal]
Hi,

got a mail with a typo in the domain. Because of that the NDR is not triggered by the receiving mailserver and thus the return path never sent a mail. I would expect that postal send an NDR to the returnpath stating that no servers are available.

<img width="675" alt="image" src="https://github.com/user-attachments/assets/7cc7c960-1fb8-4417-b6b7-c7aa5c5c1902">

Am I missing something in my config?

br,

**Comment by willpower232** [comment]
Bounces and fails are completely different things.

A bounce is when an email is fully accepted by the recipient email server but is not actually deliverable.

A hard fail is when the message could not be sent after many attempts.

A soft fail is when the message could not be sent but Postal is likely trying again.

**Comment by odoo-mvds** [comment]
I'm fully aware of that. But isn't it logic to alart the sending party if a mail hard fails?

A bounce : receiving server can send a reply to the return path mail adress with an error the mailbox doesn't exist. This is working fine with postal.

Hard bounce (no mx records) then the sending mail server should send a reply to the return path (if you use SMTP of O365, Gmail, Kerio or whatever mail server you use, you get that NDR message from your own server. However postal is not doing this.
Because of this the sender isn't aware of hard bounces.

br

**Reply by willpower232** [reply]
You can implement webhooks so that your system is aware of whether the message fails or bounces https://docs.postalserver.io/developer/webhooks

**Reply by odoo-mvds** [reply]
We also know that :) But unfortunately we are using cloud services like Odoo Online which do not allow customisation. They do process NDR message to calculate the number of bounced messages.

Are you part of the dev team of postal.io? Or is there a way we can suggest this feature to be added or even create a PR for this?

br

**Reply by willpower232** [reply]
I believe PRs are always welcome, I don't know enough Ruby myself to contribute other than poking things occasionally.

I guess worst case scenario for you is writing some middleware to receive Postals webhooks and map them into your service of choice.

**Comment by odoo-mvds** [comment]
Hi,

any idea where I can contact the team to ask if this would ever be on the roadmap?
We found a workaround by catching the webhook via make.com and send us a mail. However we are losing information like the number of bounces on mailings. Which would be the best scenario:

<img width="845" alt="image" src="https://github.com/user-attachments/assets/9b84500e-d8c8-4f76-8d2d-1a3584d7d711">
